### PR TITLE
Add check to the merge function in heading and paragraph blocks

### LINKED ISSUE #12616: "null" appears after you paste multiple blocks into a heading then press delete
Steps to reproduce:

1. Add two paragraphs.
1. Select all and cut.
1. Start a new heading.
1. Paste.

<strong>Result:</strong> the word "null" appears at the beginning of the pasted content.

screen shot 2018-12-05 at 8 37 29 am

Video: https://gfycat.com/ZigzagFlawedIchthyostega /hat tip marctison75 for the report and video posted at https://wordpress.slack.com/archives/C02QB2JS7/p1544019487258300

Video: [25s](https://cloudup.com/c5pWbKgNafS)
<sup>Tested with WordPress 5.0-RC2-43958 and also with 4.6.1 `master` @ `874321773` using Firefox 63.0.3 on macOS 10.13.6.</sup>

### PR DESCRIPTION
**Pull request by brentswisher**
## Description
If trying to merge a blank heading or paragraph, if this check isn't performed, the result is the string 'null' appearing in the text. Fixes #12616

## How has this been tested?
Start a post, create two paragraphs, select them, and cut them to the clipboard.
Start a new post, insert a heading.
*Don't type anything*
Paste the two paragraph blocks, which will insert them after the heading.
With the cursor at the start of the first paragraph, press delete (Backspace on a PC?)
See that the paragraph text is added to the header, without the word "null"
Click into a new paragraph at the bottom.
Again without typing anything, press delete.
See that you are moved to the previous paragraph, but "null" was not appended

*The key to testing this is to paste the blocks without typing anything while the cursor is still in the heading.*

A helpful video of the error happening was provided in the original issue: [https://gfycat.com/ZigzagFlawedIchthyostega](https://gfycat.com/ZigzagFlawedIchthyostega)

## Screenshot
bkm1uuhq7r

## Types of changes
Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by obenland — APPROVED** [review]
These changes do fix the bug. I'd defer to a code owner on if this is the most preferred way of adding fallbacks.

**Comment by gziolo** [comment]
@brentswisher - this is good to go. I noticed that there are more blocks which could use the same fallback technique. Do you want to include them in this PR as well?

**Review by gziolo — APPROVED** [review]

**Pull request merged**
